projects/packages/changelogger/src/Utils.php

```diff
@@ -143,11 +146,49 @@ public static function loadChangeFile( $filename, &$diagnostics = null ) {
 			throw $ex;
 		}
 		$diagnostics['lines'][''] = $line + strspn( $contents, "\n" );
-		$ret['']                  = trim( $contents );
 
+		$contents = trim( $contents );
+		if ( ! empty( $input_options['add-pr-num'] ) && $contents && $output && $formatter ) {
+			$pr_num = self::getPrNumForFile( $filename, $output, $formatter );
+			if ( $pr_num ) {
+				$contents = $contents . " [#$pr_num]";
+			}
+		}
+
+		$ret[''] = $contents;
 		return $ret;
 	}
 
+	/**
+	 * Try to get a GitHub PR number from the latest commit message subject for the given file.
+	 *
+	 * GitHub's 'squash merging' for pull requests adds the PR number by default to the end of the commit subject.
+	 * Expects the commit subject string to end like `(#123)`.
+	 *
+	 * @param string               $file Filepath.
+	 * @param OutputInterface      $output OutputInterface to write debug output to.
+	 * @param DebugFormatterHelper $formatter Formatter to use to format debug output.
+	 * @return string The PR number string if found, or an empty string.
+	 */
+	public static function getPrNumForFile( $file, OutputInterface $output, DebugFormatterHelper $formatter ) {
+		try {
+			$process = self::runCommand( array( 'git', 'log', '-1', '--format=%s', $file ), $output, $formatter );
```

Comment:
This is basically the same call as in `getTimestamp()`. We could combine these into one "getRepoData" function easily enough so we can have just one call.

For example, pass `"--format=%cI\n%s"`, then explode the result into lines and check the first line for the timestamp and the second for ending with the PR number. Or do `'--format=%cI %s'` and adjust the regex used by getTimestamp accordingly.

The "getRepoData" function would return an array with the timestamp and the PR number (if any), of course.

Reply:
Thanks for the review, I've given this a shot in b7c692805bb3a59f80577a7b5f4b4058b40da938